fix a typo in ruby type of mount resource's supports

t2y opened the pull request:
See https://github.com/chef/chef/blob/master/lib/chef/resource/mount.rb#L45

t2y committed:
fix a typo in ruby type of mount resource's supports

burtlo commented on chef_master/source/resource_mount.rst:
```diff
@@ -239,9 +239,9 @@ This resource has the following properties:
+   **Ruby Type:** Hash
 
-   An array of options for supported mount features. Default value: ``:remount``.
+   Specify a Hash of supported mount features. Default value: ``remount: true``.
```
Looking at the code and the comment above the default value, when not specifying a Hash of values, is `remount: false`.

t2y replied:
Oh, I fixed.

burtlo commented on chef_master/source/resource_mount.rst:
```diff
@@ -41,7 +41,7 @@ The full syntax for all of the properties that are available to the **mount** re
      provider                   Chef::Provider::Mount
      subscribes                 # see description
-     supports                   Array
+     supports                   Hash # defaults to { :remount => false }
```
The interesting thing about the code is that `supports` does in fact support an `Array` as well. When an `Array` is specified the resulting `Hash` are all of the keys defined in the `Array` with values set to `true`.

burtlo replied:
This mount resource using a `Hash`:
```
mount '/tmp' do
  supports { remount: true }
end
```
is equivalent to using an `Array`
```
mount '/tmp' do
  supports [ :remount ]
end
```

t2y replied:
Hmm, it's true, but it's not appropriate.

burtlo replied:
I wanted to add some comments in here around the code based on a conversation I was having someone at Chef. Hash all you want. Array all you want. 😄 I like seeing how code works and talk about it.

burtlo commented:
I think this is good change to the docs. 👍 

Also, it took me a moment: the link that you provided above now points to a different line of code. Did you mean: https://github.com/chef/chef/blob/987bcb5dd740637693dbb6723e8730f32da43696/lib/chef/resource/mount.rb#L34-L35?

t2y commented:
> Also, it took me a moment: the link that you provided above now points to a different line of code. Did you mean: https://github.com/chef/chef/blob/987bcb5dd740637693dbb6723e8730f32da43696/lib/chef/resource/mount.rb#L34-L35?

Yes, exactly.

t2y commented:
This is the source code at those days when I sent this PR.

* https://github.com/chef/chef/blob/937faa7e42c6df03b1deec0e8bbc785b7ba33e90/lib/chef/resource/mount.rb#L45

Commit:
Adds Array to supports

Adds that supports will support an array and indicates that hashes are preferred.

t2y commented:
@burtlo @kagarmoe The review of my changes has finished. I think there's no task for me. Is it right?

kagarmoe commented:
![gif-keyboard-4234956650618166531](https://cloud.githubusercontent.com/assets/4400151/24019236/812ff1a2-0a54-11e7-894a-cc5af0560e6d.gif)
